components/brave_wallet/browser/brave_wallet_utils_unittest.cc

```diff
@@ -1187,4 +1187,94 @@ TEST(BraveWalletUtilsUnitTest, GetSnsRpcUrl) {
   EXPECT_EQ(GURL("https://mainnet-beta-solana.brave.com/rpc"), GetSnsRpcUrl());
 }
 
+TEST(BraveWalletUtilsUnitTest, GetChainIdByNetworkId) {
+  TestingPrefServiceSimple prefs;
+  prefs.registry()->RegisterDictionaryPref(kBraveWalletCustomNetworks);
+  prefs.registry()->RegisterBooleanPref(kSupportEip1559OnLocalhostChain, false);
+
+  for (auto coin :
+       {mojom::CoinType::ETH, mojom::CoinType::FIL, mojom::CoinType::SOL}) {
+    ASSERT_TRUE(GetAllCustomChains(&prefs, coin).empty());
+
+    std::vector<base::Value::Dict> values;
+    mojom::NetworkInfo chain1 = GetTestNetworkInfo1();
+    chain1.coin = coin;
+    values.push_back(NetworkInfoToValue(chain1));
+
+    mojom::NetworkInfo chain2 = GetTestNetworkInfo2();
+    chain2.coin = coin;
+    if (coin != mojom::CoinType::ETH) {
+      chain2.is_eip1559 = false;
+    }
+    values.push_back(NetworkInfoToValue(chain2));
+    UpdateCustomNetworks(&prefs, std::move(values), coin);
+  }
+
+  auto chainIdNotExist =
+      GetChainIdByNetworkId(&prefs, mojom::CoinType::SOL, "testnet");
+  ASSERT_TRUE(chainIdNotExist.has_value());
+  EXPECT_EQ("0x66", chainIdNotExist.value());
+
+  chainIdNotExist =
+      GetChainIdByNetworkId(&prefs, mojom::CoinType::ETH, "goerli");
+  ASSERT_TRUE(chainIdNotExist.has_value());
+  EXPECT_EQ("0x5", chainIdNotExist.value());
+
+  chainIdNotExist =
+      GetChainIdByNetworkId(&prefs, mojom::CoinType::ETH, "mainnet");
+  ASSERT_TRUE(chainIdNotExist.has_value());
+  EXPECT_EQ("0x1", chainIdNotExist.value());
+
+  chainIdNotExist =
+      GetChainIdByNetworkId(&prefs, mojom::CoinType::ETH, "0x4e454152");
+  ASSERT_TRUE(chainIdNotExist.has_value());
+  EXPECT_EQ("0x4e454152", chainIdNotExist.value());
+
+  chainIdNotExist =
+      GetChainIdByNetworkId(&prefs, mojom::CoinType::ETH, "chain_id");
+  ASSERT_TRUE(chainIdNotExist.has_value());
+  EXPECT_EQ("chain_id", chainIdNotExist.value());
+
+  chainIdNotExist =
+      GetChainIdByNetworkId(&prefs, mojom::CoinType::ETH, "chain_id2");
+  ASSERT_TRUE(chainIdNotExist.has_value());
+  EXPECT_EQ("chain_id2", chainIdNotExist.value());
+
+  chainIdNotExist =
+      GetChainIdByNetworkId(&prefs, mojom::CoinType::SOL, "mainnet");
+  ASSERT_TRUE(chainIdNotExist.has_value());
+  EXPECT_EQ("0x65", chainIdNotExist.value());
+
+  chainIdNotExist =
+      GetChainIdByNetworkId(&prefs, mojom::CoinType::SOL, "testnet");
+  ASSERT_TRUE(chainIdNotExist.has_value());
+  EXPECT_EQ("0x66", chainIdNotExist.value());
```

Comment:
What's the difference between these and line 1213-1216?

Reply:
fixed